Added mistel/airone_pro

galican opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->
Added keyboard mistel/airone_pro with jis variant.
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
Add keyboard mistel/airone_pro with jis variant to qmk repository and request for review, Thanks!
## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

galican committed:
Added keyboard 8s200

galican committed:
8s200: Updated to new pattern

galican committed:
Refacted some codes

galican committed:
Merge pull request #3 from galican/hfdkb_8s200

Added 8s200

galican committed:
Initialize the keyboard_config to zero

galican committed:
Merge pull request #4 from galican/hfdkb_8s200

Initialize the keyboard_config to zero

waffle87 reviewed (commented):
Can you please provide a link to where this keyboard can be purchased, or where more information can be found?

galican committed:
Added picture and link for 8s200

galican committed:
Merge pull request #5 from galican/hfdkb_8s200

Added picture and link for 8s200

galican committed:
Updated the picture link

galican committed:
Back to origional picture link

galican committed:
Updated the picture link

galican committed:
Merge pull request #6 from galican/hfdkb_8s200

Update the picture link

galican committed:
Merge branch 'qmk:master' into master

drashna commented on keyboards/hfdkb/8s200/8s200.h:
```diff
@@ -0,0 +1,23 @@
+#include "quantum.h"
+
+enum user_keycodes {
+    SW_OS1 = SAFE_RANGE,
```
```suggestion
    SW_OS1 = QK_KB,
```

galican committed:
Update keyboards/hfdkb/8s200/8s200.c

Co-authored-by: Drashna Jaelre <drashna@live.com>

galican committed:
Update keyboards/hfdkb/8s200/8s200.h

Co-authored-by: Drashna Jaelre <drashna@live.com>

drashna commented on keyboards/hfdkb/8s200/keyboard.json:
```diff
@@ -0,0 +1,224 @@
+            "solid_reactive": true
+        },
+        "default": {
+            "speed": 141
```
```suggestion
            "hue": 170,
            "speed": 141
```

zvecr commented on keyboards/hfdkb/8s200/8s200.c:
```diff
@@ -0,0 +1,376 @@
+#endif
+
+// clang-format on
+typedef union {
```
Please remove all the "blink" logic, as it's not something we will want to accept.

galican replied:
I wil delete them and keep things clean enough.

zvecr commented on keyboards/hfdkb/8s200/8s200.h:
This file should be removed, and the `PDF(x) ` keycode used instead.

drashna commented on keyboards/hfdkb/8s200/8s200.c:
```diff
@@ -0,0 +1,376 @@
+}
+
+void eeconfig_init_kb() {
+    rgb_matrix_config.hsv.h = 170;
```
```suggestion
```

zvecr commented on keyboards/hfdkb/8s200/config.h:
```diff
@@ -0,0 +1,43 @@
+
+/* I2C Config for LED Driver */
+#define IS31FL3733_DRIVER_COUNT 2
+#define IS31FL3733_I2C_ADDRESS_1 0b1110100
+#define IS31FL3733_I2C_ADDRESS_2 0b1110111
```
These do not look like valid address per the datasheet, nor are they using the correct defines.

This is only a guess at A6 being incorrectly set to 1 instead of 0.....
```suggestion
#define IS31FL3733_I2C_ADDRESS_1 IS31FL3733_I2C_ADDRESS_SCL_GND
#define IS31FL3733_I2C_ADDRESS_2 IS31FL3733_I2C_ADDRESS_SCL_VCC
#define IS31FL3733_SDB_PIN A15


```

galican replied:
Sorry, I mixed up the LED control IC. According to the actual PCB, it’s actually using the SNLED27351, so its actual address starts with 0x7.

zvecr commented on keyboards/hfdkb/8s200/config.h:
```diff
@@ -0,0 +1,43 @@
+#define IS31FL3733_DRIVER_COUNT 2
+#define IS31FL3733_I2C_ADDRESS_1 0b1110100
+#define IS31FL3733_I2C_ADDRESS_2 0b1110111
+#define I2C1_SDA_PIN B7
+#define I2C1_SCL_PIN B6
+#define I2C1_SCL_PAL_MODE 4
+#define I2C1_OPMODE OPMODE_I2C
```
Duplicates defaults
```suggestion
```

galican replied:
If #define I2C1_OPMODE OPMODE_I2C isn’t defined, compilation will fail. Tracing the I2C1_OPMODE macro shows that USE_I2CV1 needs to be defined first in i2c_master.c. Following USE_I2CV1 further, it turns out that for WB32 series MCUs, USE_I2CV1 isn’t defined in chibios_config.h. As a result, using the default value leads to compilation errors. Other keyboards using WB32 series MCUs manually define I2C1_OPMODE in their config.h to avoid this issue. Please review again when it's convenient for you, Thank.

galican committed:
Update keyboards/hfdkb/8s200/8s200.c

Co-authored-by: Drashna Jaelre <drashna@live.com>

galican committed:
Update keyboards/hfdkb/8s200/config.h

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/keyboard.json

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/config.h

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/config.h

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/config.h

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/config.h

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/keyboard.json

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/keyboard.json

Co-authored-by: Drashna Jaelre <drashna@live.com>

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Removed 8s200.h.
Replaced the rbg matrix deriver with snled27351 according to actual schematic diagram.
Removed all ''blink' and "rgb_matrix_effect_table" logic.
Removed unused pin definiton like RGB_DRIVER_SDB_PIN and RGB_DRIVER_EN_PIN.

galican committed:
Merge pull request #10 from galican/master

Removed 8s200.h.

galican commented on keyboards/hfdkb/8s200/8s200.c:
```diff
@@ -0,0 +1,376 @@
+    writePinHigh(RGB_DRIVER_SDB_PIN);
+#endif
+    matrix_init_user();
+}
```
RGB_DRIVER_SDB_PIN is no longer needed, An external pull-up has already been added to the SDB pin in the circuit, so it will always be enabled. please review again when it's convenient for you, Thanks.

galican committed:
Update keyboards/hfdkb/8s200/8s200.c

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Update keyboards/hfdkb/8s200/keyboard.json

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Merge branch 'qmk:master' into master

zvecr commented on keyboards/hfdkb/8s200/8s200.c:
```diff
@@ -0,0 +1,162 @@
+
+// clang-format on
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (process_record_user(keycode, record) != true) {
```
Nitpick, but aligns to how we generally have these callbacks implemented
```suggestion
    if (!process_record_user(keycode, record)) {
```

galican committed:
Update keyboards/hfdkb/8s200/8s200.c

All right, thanks~

Co-authored-by: Joel Challis <git@zvecr.com>

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

tzarc commented on keyboards/hfdkb/8s200/readme.md:
```diff
@@ -0,0 +1,27 @@
+
+* Keyboard Maintainer: [jonylee@hfd](https://github.com/jonylee1986)
+* Hardware Supported: 8s200
+* Hardware Availability: [Mistel AIRONE PRO Black-JIS keyboard](https://mistelkeyboard.com/products/2a25e7752bf9a3d8e66548816250028a)
```
The entire board should be under `keyboards/mistel/airone_pro`. `hfdkb/8s200` means nothing to end-users.

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Added mistel/airone_pro.

galican committed:
Updated mistel/airone_pro/jis.

galican committed:
Merge pull request #16 from galican/master

Added mistel/airone_pro.

galican committed:
Updated readme.md.

galican committed:
Merge branch 'jonylee1986:master' into master

galican committed:
Merge pull request #17 from galican/master

Updated readme.md of mistel/airone_pro.

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master

galican committed:
Merge branch 'qmk:master' into master
